### PR TITLE
DistriOptimizerV2 argument

### PR DESCRIPTION
**Pull request by Le-Zheng**

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/Engine.scala [hunk 1]
```diff
@@ -122,6 +122,7 @@ object Engine {
   private val singletonCounter = new AtomicBoolean()
   private var physicalCoreNumber = -1
   private var nodeNum: Int = -1
+  private var optimizerVersion: Boolean = false
```
Comment (qiuxin2012) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/Engine.scala:125]: You can copy some code from `engineType`

**Comment by Le-Zheng** [comment]
To fix https://github.com/intel-analytics/BigDL/issues/2999

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/Engine.scala [hunk 2]
```diff
@@ -37,6 +37,14 @@ sealed trait EngineType
 case object MklBlas extends EngineType
 case object MklDnn extends EngineType
 
+/**
+ * define optimizer version trait
+ */
+sealed trait OptimizerVersion
+
+case object Optimizer1 extends OptimizerVersion
+case object Optimizer2 extends OptimizerVersion
```
Comment (jason-dai) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/Engine.scala:46]: change to OptimizerV1 and OptimizerV2

**Comment by jason-dai** [comment]
> To fix #2999

This only fixes the first task in #2999

**Comment by Le-Zheng** [comment]
Jenkins related tests passed http://10.239.47.210:18080/job/BigDL-PRVN-lite/1143/

**Pull request merged**